pkg/platform/k8s/pod_controller_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"context"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	TestPodName      = "test-pod"
+	TestPodNameSpace = "default"
+	TestPodKeyLabel  = "app"
+	TestPodIP        = "10.0.0.1"
+)
+
+func TestPodReconciler(t *testing.T) {
+	// Test setup
+	logger := logrus.WithField("component", "podReconciler")
+	client, err := getFakeClient()
+	require.NoError(t, err)
+	ctx := context.Background()
+	podReconciler := &PodReconciler{
+		Client:  client,
+		ipToPod: make(map[string]metav1.ObjectMeta),
+		logger:  logger,
+	}
+	req := ctrl.Request{NamespacedName: types.NamespacedName{
+		Name:      TestPodName,
+		Namespace: TestPodNameSpace,
+	}}
+
+	// Pod creation check
+	createLabel := "create-label"
+	pod := getFakePod(createLabel)
```

Comment:
are you adding the Pod object to the fake client or creating it?
I'm not clear as to what the right approach would be, but I would expect the only methods exercised on the reconciler are the constructor and then reconcile() followed by lookup.

Reply:
I am trying to check just the reconcile logic, so I add or delete pods using the fake client and then call the reconcile to check the logic for adding pod info.